ui/app/AppLayouts/Onboarding/views/InsertDetailsView.qml

```diff
@@ -0,0 +1,264 @@
+                icon.charactersLen: 2
+                image.isIdenticon: false
+                image.source: uploadProfilePicPopup.selectedImage
+                ringSettings { ringSpecModel: Utils.getColorHashAsJson(root.pubKey) }
+            }
+            StatusRoundButton {
+                id: updatePicButton
+                width: 40
+                height: 40
+                anchors.top: parent.top
+                anchors.right: parent.right
+                type: StatusFlatRoundButton.Type.Secondary
+                icon.name: "add"
+                onClicked: {
+                    uploadProfilePicPopup.open();
+                }
+            }
+        }
+
+        StatusInput {
+            id: nameInput
+            implicitWidth: 328
+            Layout.alignment: Qt.AlignHCenter
+            input.placeholderText: qsTr("Display name")
+            input.edit.font.capitalization: Font.Capitalize
+            input.component: RoundedIcon {
+                width: 14
+                height: 14
+                iconWidth: 14
+                iconHeight: 14
+                color: "transparent"
+                source: Style.svg("close-filled")
+                onClicked: {
+                    nameInput.input.edit.clear();
+                }
+            }
+            onTextChanged: {
+                userImage.name = text;
+            }
+        }
+
+        StyledText {
+            id: chatKeyTxt
+            color: Style.current.secondaryText
+            text: "Chatkey:" + root.address
+            horizontalAlignment: Text.AlignHCenter
+            wrapMode: Text.WordWrap
+            Layout.alignment: Qt.AlignHCenter
+            font.pixelSize: 15
+        }
+
+        Item {
+            id: chainsChatKeyImg
+            Layout.alignment: Qt.AlignHCenter
+            Layout.preferredWidth: 181
+            Layout.preferredHeight: 84
+            Image {
+                anchors.horizontalCenter: parent.horizontalCenter
+                source: Style.png("onboarding/chains")
+            }
+            Text {
+                anchors.bottom: parent.bottom
+                text: {
+                    const emojiHash = Utils.getEmojiHashAsJson(root.pubKey);
+                    var emojiHashFirstLine = "";
+                    var emojiHashSecondLine = "";
+                    for (var i = 0; i < 7; i++) {
+                        emojiHashFirstLine += emojiHash[i];
+                    }
+                    for (var j = 7; j < emojiHash.length; j++) {
+                        emojiHashSecondLine += emojiHash[j];
+                    }
+                    return !!StatusQUtils.Emoji.parse(emojiHashFirstLine, "14x14") + "<br>" + StatusQUtils.Emoji.parse(emojiHashSecondLine, "14x14") ?
+                             StatusQUtils.Emoji.parse(emojiHashFirstLine, "14x14") + "<br>" + StatusQUtils.Emoji.parse(emojiHashSecondLine, "14x14") :
+                             "";
+                }
+            }
```

Comment:
That's code duplication, please extract component out of that.

Reply:
done, created EmojiHash in shared.controls and replaced here as well as in ProfileHeader